MTGSDKSwift/MagicalCardManager.swift

```diff
@@ -8,74 +8,127 @@
 
 import UIKit
 
-
 final public class Magic {
-    public init() {}
+    public typealias CardImageCompletion = (Result<UIImage>) -> Void
+    public typealias CardCompletion = (Result<[Card]>) -> Void
+    public typealias SetCompletion = (Result<[CardSet]>) -> Void
 
-    public typealias CardImageCompletion = (UIImage?, NetworkError?) -> Void
-    public typealias CardCompletion = ([Card]?, NetworkError?) -> Void
-    public typealias SetCompletion = ([CardSet]?, NetworkError?) -> Void
+    private let mtgAPIService: MTGAPIService
+    
+    public init() {
+        self.mtgAPIService = MTGAPIService()
+    }
+    
+    /**
+     Reteives an array of Cards which match the parameters given
+     - parameter parameters: [CardSearchParameter]
+     - parameter configuration: MTGSearchConfiguration
+     - parameter completion: (Result<[Card]>) -> Void
+     */
+    public func fetchCards(_ parameters: [CardSearchParameter],
+                           _ configuration: MTGSearchConfiguration = MTGSearchConfiguration.defaultConfiguration,
+                           _ completion: @escaping CardCompletion) {
+        var networkError: NetworkError? {
+            didSet {
+                completion(Result.error(networkError!))
+            }
+        }
+        
+        guard let url = URLBuilder.buildURLWithParameters(parameters, andConfig: configuration) else {
+            networkError = NetworkError.miscError("fetchCards url build failed")
+            return
+        }
+        
+        mtgAPIService.mtgAPIQuery(url: url) {
+            result in
+            switch result {
+            case .success(let json):
+                let cards = Parser.parseCards(json: json)
+                completion(Result.success(cards))
+            case .error(let error):
+                networkError = error
+            }
+        }
+    }
     
-    public static var enableLogging: Bool = false
-    public var fetchPageSize: String = "12"
-    public var fetchPageTotal: String = "1"
+    /**
```

Comment:
Can we switch this to the modern comment syntax (`///`)?  Thanks for adding comments!